src/main/java/sirius/db/mixing/properties/LongProperty.java

```diff
@@ -106,4 +107,10 @@ public void describeProperty(JSONObject description) {
                        IndexMode::docValues,
                        description);
     }
+
+    @Override
+    protected void onBeforeSaveChecks(Object entity) {
+        super.onBeforeSaveChecks(entity);
+        assertValueIsInRange(Amount.of((Long) getValue(entity)));
```

Comment:
couldn't this be generalized? Value.of(getValue(entity)).getAmount()

Reply:
+1 it would be nice if `NumberProperty`  would be in charge of calling `assertValueIsInRange` and not every subclass separately